torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.h

```diff
@@ -438,24 +469,96 @@ class TORCH_CUDA_CU_API ContiguousInnerDimensionsMapper
  private:
   ContiguousInnerDimensionsMapper(
       TensorView* reference,
-      std::vector<IterDomain*> ids,
+      const std::vector<IterDomain*>& reference_ids,
       std::shared_ptr<const ComputeAtMap> ca_map,
       const std::unordered_set<Split*>& divisible_splits);
 
-  std::unordered_map<TensorView*, std::vector<IterDomain*>>::iterator
-  projectIdToRoot(TensorView* ref, std::vector<IterDomain*> ids);
+  class MappedDomain : public Information {
+   public:
+    MappedDomain() = default;
+
+    static std::shared_ptr<MappedDomain> build(
+        std::vector<IterDomain*> root_ids,
+        std::vector<IterDomain*> rfactor_ids) {
+      auto ptr = std::make_shared<MappedDomain>();
+      ptr->mapped_root_ids_ = root_ids;
+      ptr->mapped_rfactor_ids_ = rfactor_ids;
+      return ptr;
+    }
+
+    operator bool() const final {
+      return !mapped_root_ids_.empty() || !mapped_rfactor_ids_.empty();
+    }
 
-  std::unordered_map<TensorView*, std::vector<IterDomain*>>::iterator
-  projectIdToRFactor(TensorView* ref, std::vector<IterDomain*> ids);
+    bool operator<(const Information& other_info) const final {
+      auto other_mapped_domain = dynamic_cast<const MappedDomain&>(other_info);
+      return mapped_root_ids_.size() >
+          other_mapped_domain.mapped_root_ids_.size() &&
+          mapped_rfactor_ids_.size() >
+          other_mapped_domain.mapped_rfactor_ids_.size();
```

Comment:
Should this be `<` instead of `>`?  In `MaxRootDomainInfoSpanningTree::Information`, we are using `<`.
https://github.com/csarofeen/pytorch/blob/2057f37541e0c3e898094d6cb8246756520853ac/torch/csrc/jit/codegen/cuda/maxinfo_propagator.cpp#L164-L176

Reply:
I'm sure you're right, but for some reason it doesn't seem to make any difference whatsoever if it's `>` or `<`. In fact I also tried just plain `true` and `false` and no matter what I pick here it seems to work as required. I can't seem to bottom out on this why, but if you want to play around with it, this is a test where you can see the mappings aren't great on the `vectorize` branch (the other vectorize PR), but works fine on this branch.

```.cpp
// Test propogation doesn't proceed across missing dimensions
TEST_F(NVFuserTest, FusionVectorizeSpanningTree_CUDA) {
  auto fusion_ptr = std::make_unique<Fusion>();
  Fusion& fusion = *fusion_ptr.get();
  FusionGuard fg(&fusion);

  auto bcast_inp = makeContigConcreteTensor({2});
  fusion.addInput(bcast_inp);
  auto bcast = broadcast(bcast_inp, {false, true});

  std::vector<TensorView*> intermediates;
  std::vector<TensorView*> outputs;
  for (auto i : c10::irange(10)) {
    auto resolution_inp = makeContigConcreteTensor({2, 2});
    fusion.addInput(resolution_inp);
    auto intermediate = add(bcast, resolution_inp);
    if (i > 0) {
      auto output = add(intermediates.back(), intermediate);
      outputs.push_back(output);
      fusion.addOutput(output);
    }
    intermediates.push_back(intermediate);
  }
    
  fusion.printMath();

  auto mapper = vectorize_helper::ContiguousInnerDimensionsMapper::map(
      outputs[0], {outputs[0]->axis(0), outputs[0]->axis(1)});

  for (auto tv : ir_utils::allTvs(&fusion)) {
    std::cout << tv->toString() << " :: ";
    for (auto axis : tv->getRootDomain()) {
      std::cout << mapper.getMappedExtent(axis).toString() << ", ";
    }
    std::cout << std::endl;
  }
}
```

It seems no matter what, we never propagate from the `bcast` `TensorView` in to anything but the input with dim 1 in this problem. Before this PR, the mappings printed from this test is:
```
T0_g[ iS0{2} ] :: {} / {}, 
T1_l[ iS1{2}, bS2{1} ] :: {} / {}, {} / {}, 
T2_g[ iS3{2}, iS4{2} ] :: {2} / {}, {2} / {}, 
T3_l[ iS5{2}, iS6{2} ] :: {2} / {}, {2} / {}, 
T4_g[ iS7{2}, iS8{2} ] :: {2} / {}, {2} / {}, 
T5_l[ iS9{2}, iS10{2} ] :: {2} / {}, {2} / {}, 
T6_g[ iS11{2}, iS12{2} ] :: {2} / {}, {2} / {}, 
T7_g[ iS13{2}, iS14{2} ] :: {} / {}, {} / {}, 
T8_l[ iS15{2}, iS16{2} ] :: {} / {}, {} / {}, 
T9_g[ iS17{2}, iS18{2} ] :: {2} / {}, {2} / {}, 
T10_g[ iS19{2}, iS20{2} ] :: {} / {}, {} / {}, 
T11_l[ iS21{2}, iS22{2} ] :: {} / {}, {} / {}, 
T12_g[ iS23{2}, iS24{2} ] :: {} / {}, {} / {}, 
T13_g[ iS25{2}, iS26{2} ] :: {} / {}, {} / {}, 
T14_l[ iS27{2}, iS28{2} ] :: {} / {}, {} / {}, 
T15_g[ iS29{2}, iS30{2} ] :: {} / {}, {} / {}, 
T16_g[ iS31{2}, iS32{2} ] :: {} / {}, {} / {}, 
T17_l[ iS33{2}, iS34{2} ] :: {} / {}, {} / {}, 
T18_g[ iS35{2}, iS36{2} ] :: {} / {}, {} / {}, 
T19_g[ iS37{2}, iS38{2} ] :: {} / {}, {} / {}, 
T20_l[ iS39{2}, iS40{2} ] :: {} / {}, {} / {}, 
T21_g[ iS41{2}, iS42{2} ] :: {} / {}, {} / {}, 
T22_g[ iS43{2}, iS44{2} ] :: {} / {}, {} / {}, 
T23_l[ iS45{2}, iS46{2} ] :: {} / {}, {} / {}, 
T24_g[ iS47{2}, iS48{2} ] :: {} / {}, {} / {}, 
T25_g[ iS49{2}, iS50{2} ] :: {} / {}, {} / {}, 
T26_l[ iS51{2}, iS52{2} ] :: {} / {}, {} / {}, 
T27_g[ iS53{2}, iS54{2} ] :: {} / {}, {} / {}, 
T28_g[ iS55{2}, iS56{2} ] :: {} / {}, {} / {}, 
T29_l[ iS57{2}, iS58{2} ] :: {} / {}, {} / {}, 
T30_g[ iS59{2}, iS60{2} ] :: {} / {}, {} / {}, 
```
So you can see a lot of mappings broke because it was propagated through T1, however on this PR, it's correctly:
```.cpp
T0_g[ iS0{2} ] :: {} / {}, 
T1_l[ iS1{2}, bS2{1} ] :: {} / {}, {} / {}, 
T2_g[ iS3{2}, iS4{2} ] :: {2} / {}, {2} / {}, 
T3_l[ iS5{2}, iS6{2} ] :: {2} / {}, {2} / {}, 
T4_g[ iS7{2}, iS8{2} ] :: {2} / {}, {2} / {}, 
T5_l[ iS9{2}, iS10{2} ] :: {2} / {}, {2} / {}, 
T6_g[ iS11{2}, iS12{2} ] :: {2} / {}, {2} / {}, 
T7_g[ iS13{2}, iS14{2} ] :: {2} / {}, {2} / {}, 
T8_l[ iS15{2}, iS16{2} ] :: {2} / {}, {2} / {}, 
T9_g[ iS17{2}, iS18{2} ] :: {2} / {}, {2} / {}, 
T10_g[ iS19{2}, iS20{2} ] :: {2} / {}, {2} / {}, 
T11_l[ iS21{2}, iS22{2} ] :: {2} / {}, {2} / {}, 
T12_g[ iS23{2}, iS24{2} ] :: {2} / {}, {2} / {}, 
T13_g[ iS25{2}, iS26{2} ] :: {2} / {}, {2} / {}, 
T14_l[ iS27{2}, iS28{2} ] :: {2} / {}, {2} / {}, 
T15_g[ iS29{2}, iS30{2} ] :: {2} / {}, {2} / {}, 
T16_g[ iS31{2}, iS32{2} ] :: {2} / {}, {2} / {}, 
T17_l[ iS33{2}, iS34{2} ] :: {2} / {}, {2} / {}, 
T18_g[ iS35{2}, iS36{2} ] :: {2} / {}, {2} / {}, 
T19_g[ iS37{2}, iS38{2} ] :: {2} / {}, {2} / {}, 
T20_l[ iS39{2}, iS40{2} ] :: {2} / {}, {2} / {}, 
T21_g[ iS41{2}, iS42{2} ] :: {2} / {}, {2} / {}, 
T22_g[ iS43{2}, iS44{2} ] :: {2} / {}, {2} / {}, 
T23_l[ iS45{2}, iS46{2} ] :: {2} / {}, {2} / {}, 
T24_g[ iS47{2}, iS48{2} ] :: {2} / {}, {2} / {}, 
T25_g[ iS49{2}, iS50{2} ] :: {2} / {}, {2} / {}, 
T26_l[ iS51{2}, iS52{2} ] :: {2} / {}, {2} / {}, 
T27_g[ iS53{2}, iS54{2} ] :: {2} / {}, {2} / {}, 
T28_g[ iS55{2}, iS56{2} ] :: {2} / {}, {2} / {}, 
T29_l[ iS57{2}, iS58{2} ] :: {2} / {}, {2} / {}, 
T30_g[ iS59{2}, iS60{2} ] :: {2} / {}, {2} / {}, 
```
No matter what choice is made for the information comparator function!!!

Reply:
Tried building out a better test that reverses ordering, and what tensor we're starting from for the mapping:
```.cpp

// Test propogation doesn't proceed across missing dimensions
TEST_F(NVFuserTest, FusionVectorizeSpanningTree_CUDA) {
  auto fusion_ptr = std::make_unique<Fusion>();
  Fusion& fusion = *fusion_ptr.get();
  FusionGuard fg(&fusion);

  std::vector<TensorView*> inputs;
  std::vector<TensorView*> intermediates;
  std::vector<TensorView*> outputs;

  auto bcast_inp = makeContigConcreteTensor({2});
  inputs.push_back(bcast_inp);
  auto bcast = broadcast(bcast_inp, {false, true});

  for (auto i : c10::irange(10)) {
    auto resolution_inp = makeContigConcreteTensor({2, 2});
    inputs.push_back(resolution_inp);
    auto intermediate = add(bcast, resolution_inp);
    if (i > 0) {
      auto output = add(intermediates.back(), intermediate);
      outputs.push_back(output);
    }
    intermediates.push_back(intermediate);
  }

  for (auto rev_inp : {false, true}) {
    for (auto rev_out : {false, true}) {
      // Clear fusion inputs / outputs
      {
        auto fusion_outs = fusion.outputs();
        for (auto out : fusion_outs) {
          fusion.removeOutput(out);
        }
        auto fusion_inps = fusion.inputs();
        for (auto inp : fusion_inps) {
          fusion.removeOutput(inp);
        }
      }

      if (rev_inp) {
        std::reverse(inputs.begin(), inputs.end());
      }

      if (rev_out) {
        std::reverse(outputs.begin(), outputs.end());
      }

      {
        // Populate outputs and inputs
        for (auto out : outputs) {
          fusion.addOutput(out);
        }

        for (auto inp : inputs) {
          fusion.addInput(inp);
        }
      }

      for (auto out : outputs) {
        auto mapper = vectorize_helper::ContiguousInnerDimensionsMapper::map(
            out, {out->axis(0), out->axis(1)});

        for (auto tv : ir_utils::allTvs(&fusion)) {
          if (tv->name() == 0 || tv->name() == 1) {
            continue;
          }
          for (auto axis : tv->getRootDomain()) {
            TORCH_INTERNAL_ASSERT(
                mapper.getMappedExtent(axis).getNumerator()->evaluateInt() ==
                2);
            TORCH_INTERNAL_ASSERT(
                mapper.getMappedExtent(axis).getDenominator()->evaluateInt() ==
                1);
          }
        }
      }
    }
  }
}
```